### ISSUE TITLE
Remove conditional use of wxWidgets?

### ISSUE THREAD

**Issue by ghost** [KeyWorksRW/ttLib#47]
### Description:
We can't build in the main directory with **wxWidgets** without changing our precompiled header, and if that change got checked in, it breaks everyone who doesn't have **wxWidgets** installed. One could create a different directory with its own **pch.h** file and the MSVC compiler would build correctly, but CLANG would fail because it would recognize that a different **pch.h** was being used. Since we don't have a method in place to build a version of **ttLib** that uses **wxWidgets**, we might consider removing all the **_WX_WX_H_** conditional code .

The only reason we have any **_WX_WX_H_** conditional code is to replace Windows API with a cross platform version. In some cases our function simply calls the **wxWidgets** equivalent. For example, **tt::CreateDir** simply calls **wxDir::Make** and returns. If the caller is already linking to **wxWidgets** they would be better off calling **wxWidgets** directly if it has a function equivalent to **ttLib**. If the caller isn't linking to **wxWdigets** then they couldn't use a wxWidgets-enabled version of **ttLib** anyway (since it would require the **wxWidgets** libraries).

I don't think relying on **wxWidgets** for a cross-platform version of **ttLib** is the best choice. Instead, we could extend the practice of putting functions and classes that can only be used in Windows in a `#ifdef _WINDOWS_` conditional section. We should use C runtime calls rather then direct calls to the Windows API when there is an equivalency. Otherwise, we'll need a dev to step in and update the code for portability issues that can be resolved without requiring **wxWidgets**.

Bottom line: I would prefer not to depend on any specific library to enhance our cross-platform usability.

### Issue Type:
- [ ] Bug report
- [ ] Feature request
- [x] Other
